dask/array/reductions.py

```diff
@@ -714,7 +716,8 @@ def cumreduction(func, binop, ident, x, axis=None, dtype=None, out=None):
 
     m = x.map_blocks(func, axis=axis, dtype=dtype)
 
-    name = '%s-axis=%d-%s' % (func.__name__, axis, tokenize(x, dtype))
+    name = '{0}-axis={1}-{2}'.format(func.__name__, axis, tokenize(func, x, dtype,
+                                                                   dtype, out))
```

Comment:
Should we include `ident` here but exclude `out`?

Reply:
including `ident` can not hurt I guess, I am not sure about leaving out out, although I see why one would not like to include it. Drop axis here as well then.

Reply:
I would, yes.  Historically it was common to put a lot of information into the key.  Over time this practice was dropped because keys got too long and no one really used this information anyway.

Reply:
For leaving out out I think it makes sense because if we do something like the following:

```python
np.add(x, y, out=a)
np.add(x, y, out=b)
```

Then presumably we want `a.name == b.name`

Reply:
But what does the following do?
```python
da.add(x, y, out=x)
da.add(x, y, out=a)
```
Does this get solved by `x` being a different array in the second call?

Reply:
It's the same semantics as Numpy.  It's technically the same `Array` object but the data (or in this case task graph) are different.

Reply:
Yes, so that means that in the above case as expected, we will end up with a=x+y+y as the `Array` object `x` will tokenize differently the second time, right?

Reply:
Correct